rusk-wallet/src/lib/cache.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+use phoenix_core::Note;
+use rusqlite::{params, Connection};
+
+pub(crate) fn open(data_dir: &PathBuf) -> Connection {
```

Comment:
I can think of a condition like the disk being temporarily inaccessible and the connection dropping for example. I would prefer some error propagation with user feedback rather than a panic.